value/src/test/java/com/google/auto/value/processor/CompilationTest.java

```diff
@@ -1678,4 +1678,145 @@ public void testReferencingGeneratedClass() {
         .processedWith(new AutoValueProcessor(), new FooProcessor())
         .compilesWithoutError();
   }
+
+  public void testHashCode() {
```

Comment:
I'd rather not have this extra test method. It is that much more code that we have to update every time we change the code template. The new AutoValueTest test case should be enough.